pkg/process/monitor/process_monitor_test.go

```diff
@@ -179,4 +214,20 @@ func TestProcessMonitorInNamespace(t *testing.T) {
 		_, captured := execSet.Load(cmd.ProcessState.Pid())
 		return captured
 	}, time.Second, 200*time.Millisecond, "did not capture process EXEC from other namespace")
+
+	tel := telemetry.ReportPayloadTelemetry("1")
```

Comment:
I don't think this is the proper way to test this.
This function will only return things instantiated with `OptPayloadTelemetry`,  and your metrics are all `OptPrometheus`.

In order to decouple your tests from the metric "export" type, I'd simply write assertions like 
```
assert.Equal(t, 1, pm.tel.callbackExecuted.Get())
```
Just make sure to call `telemetry.Clear()` at the beginning of tests because these counters are monotonic

Reply:
I added a return to ReportPrometheus()
This way I would check end to end the value are correct with the correct exported name